nixos/tests/pgadmin4.nix

```diff
@@ -5,62 +5,93 @@ import ./make-test-python.nix ({ pkgs, lib, ... }:
     pgadmin4Dir = "/var/lib/pgadmin";
     pgadmin4LogDir = "/var/log/pgadmin";
 
-    python-with-needed-packages = pkgs.python3.withPackages (ps: with ps; [
-      selenium
-      testtools
-      testscenarios
-      flask
-      flask-babelex
-      flask-babel
-      flask-gravatar
-      flask_login
-      flask_mail
-      flask_migrate
-      flask_sqlalchemy
-      flask_wtf
-      flask-compress
-      passlib
-      pytz
-      simplejson
-      six
-      sqlparse
-      wtforms
-      flask-paranoid
-      psutil
-      psycopg2
-      python-dateutil
-      sqlalchemy
-      itsdangerous
-      flask-security-too
-      bcrypt
-      cryptography
-      sshtunnel
-      ldap3
-      gssapi
-      flask-socketio
-      eventlet
-      httpagentparser
-      user-agents
-      wheel
-      authlib
-      qrcode
-      pillow
-      pyotp
-      boto3
-    ]);
   in
   {
     name = "pgadmin4";
     meta.maintainers = with lib.maintainers; [ gador ];
 
     nodes.machine = { pkgs, ... }: {
       imports = [ ./common/x11.nix ];
+      # needed because pgadmin 6.8 will fail, if those dependencies get updated
+      nixpkgs.overlays = [
+        (self: super: {
+          pythonPackages = super.python3.pkgs.overrideScope (final: prev: rec {
+
+            flask = prev.flask.overridePythonAttrs (oldAttrs: rec {
+              version = "2.0.3";
+              src = oldAttrs.src.override {
+                inherit version;
+                sha256 = "sha256-4RIMIoyi9VO0cN9KX6knq2YlhGdSYGmYGz6wqRkCaH0=";
+              };
+              disabledTests = (oldAttrs.disabledTests or [ ]) ++ [
+                "test_aborting"
+              ];
+            });
+            flask-paranoid = prev.flask-paranoid.overridePythonAttrs (oldAttrs: rec {
+              # Nothing of interest changed from 0.2 to 0.3
+              doCheck = false;
+            });
+            werkzeug = prev.werkzeug.overridePythonAttrs (oldAttrs: rec {
+              version = "2.0.3";
+              src = oldAttrs.src.override {
+                inherit version;
+                sha256 = "sha256-uGP4/wV8UiFktgZ8niiwQRYbS+W6TQ2s7qpQoWOCLTw=";
+              };
+            });
+          });
+        })
+      ];
+
       environment.systemPackages = with pkgs; [
         pgadmin4
         postgresql
-        python-with-needed-packages
         chromedriver
         chromium
+        (python3.withPackages
```

Comment:
why is this environment needed? pgadmin should be wrapped with it

Reply:
Yes, Pgadmin is wrapped.
But the tests (which are executed here, because they need a postgres server running) are not.
Also, Pgadmin is not wrapped with test python packages (such as testscenarios). 

I tried all kinds of things, but this is the only working solution I have found, which overrides the packages.
If you have any suggestions of how else to implement the regression tests, I'm open to them 🙂

Reply:
It's probably cleaner to put take the various *inputs then from `pgadmin4` and build a environment using those. No way to repeat packages that way.

Reply:
Yes, i totally agree. Unfortunately, I can't seem to get it to work.

What I tried:
- adding `pkgs.pgadmin4.propagatedBuildInputs` to  `environment.systemPackages = with pkgs; [ ]`
- adding `pkgs.pgadmin4.propagatedBuildInputs` to `(python3.withPackages (ps: with pythonPackages; [ ] `
- trying to instigate a `nix-shell  -p pgadmin` within the VM (which would have all those dependencies linked)

At this point I am at a loss. I really don't like repeating the packages from the `pgadmin4` derivation, on the other hand 
this is the only way I got it to work.
I have looked at other `nixos/tests` examples but didn't find anything related.

Do you have any suggestions?

Reply:
I found a way.